apps/src/templates/feedback/Code.jsx

```diff
@@ -0,0 +1,17 @@
+var msg = require('../../locale');
+
+var Code = React.createClass({
+  
+  propTypes: {
+    message: React.PropTypes.string.isRequired,
+    code: React.PropTypes.string.isRequired,
+  },
+
+  render: function () {
+    return (<div className="generated-code-container">
+      <p className="generatedCodeMessage" dangerouslySetInnerHTML={{__html: this.props.message}} />
+      <pre className="generatedCode" dangerouslySetInnerHTML={{ __html: this.props.code }} />
+    </div>);
```

Comment:
It seems like the reason React is so paranoid about rendering unescaped HTML is that it assumes you're doing this primarily in situations where you're rendering user-created content, and that in all other situations you should just render another React component.

It seems like we primarily use dangerouslySetInnerHTML in situations where we want to render pre-rendered content that had been generated by some other non-react part of our system, most often stuff that comes from levelbuilder. It's sort of a necessary evil as long as our system is not 100% React.

(note in this case, the `this.props.code` element actually IS rendering user-created content, albiet some heavily processed and controlled content).

We can replace `generatedCode` with an inline style, but `generatedCodeMessage` is using the class to style some of that `dangerouslySetInnerHTML` content.